trin-core/src/types/header.rs

```diff
@@ -180,6 +182,59 @@ impl PartialEq for Header {
     }
 }
 
+/// Datatype for use in bridge functionality. The purpose is a single datatype that can be created
+/// from a header and contains all the information necessary to build history network content
+/// values for BlockBodies (txs, uncles) and Receipts (tx_hashes) through subsequnt jsonrpc
+/// requests.
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub struct FullHeader {
+    pub header: Header,
+    pub txs: Vec<Transaction>,
+    pub tx_hashes: TxHashes,
+    pub uncles: Vec<H256>,
+}
+
+// Prefer TryFrom<Value> over implementing Deserialize trait, since it's much simpler when
```

Comment:
I'm not sure if I agree with this. I think adding a lot of  `TryFrom` implementations build up with time and it becomes very hard to follow all the logic. I don't like the pattern of first converting json field to `Value` and then converting the Value to a type. I guess a more good practice is to directly convert a json string to type using Serialize/Deserialize.

If we have a json object which consists of multiple types, I think we should create a rust struct for the whole json object instead of implementing structs only for the subtypes and then doing some fancy conversion field by field to a Value type and then converting from Value to a type struct.